.github/workflows/mainPipeline.yml

```diff
@@ -0,0 +1,164 @@
+          restore-keys: ${{ runner.os }}-sonar
+          
+      - name: Cache SonarCloud scanner
+        id: cache-sonar-scanner
+        uses: actions/cache@v1
+        with:
+          path: .\.sonar\scanner
+          key: ${{ runner.os }}-sonar-scanner
+          restore-keys: ${{ runner.os }}-sonar-scanner
+          
+      - name: Install SonarCloud scanner
+        if: steps.cache-sonar-scanner.outputs.cache-hit != 'true'
+        shell: powershell
+        run: |
+          New-Item -Path .\.sonar\scanner -ItemType Directory
+          dotnet tool update dotnet-sonarscanner --tool-path .\.sonar\scanner
+          
+      - name: Run tests
+        shell: powershell
+        run: dotnet test Selenium.Axe/Selenium.Axe.Test/Selenium.Axe.Test.csproj --logger trx --results-directory "${{ github.workspace }}/artifactTests/coverage/testResults/" /p:CoverletOutput="${{ github.workspace }}/artifactTests/coverage/testResults/" /p:CollectCoverage=true /p:CoverletOutputFormat=opencover -c Release -v D
+      
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        shell: powershell
+        run: | 
+          .\.sonar\scanner\dotnet-sonarscanner begin /k:"TroyWalshProf_SeleniumAxeDotnet" /o:"troywalshprof"  /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:projectVersion="${{ steps.ver.outputs.BUILD_NUMBER }}"  /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.opencover.reportsPaths="${{ github.workspace }}/artifactTests/coverage/testResults/*.opencover.xml"  /d:sonar.cs.dotcover.reportsPaths="${{ github.workspace }}/artifactTests/coverage/testResults/*.coverage"
+          dotnet build Selenium.Axe/Selenium.Axe.sln  --configuration Release -p:Version=${{ steps.ver.outputs.BUILD_NUMBER }} -o "${{github.workspace}}/artifactTests/packages"
+          .\.sonar\scanner\dotnet-sonarscanner end /d:sonar.login="${{ secrets.SONAR_TOKEN }}"
+ 
+      - uses: actions/upload-artifact@v2
+        if: always()
+        name: "Upload NuGet packages artifact"
+        with:
+          name: packages
+          path: |
+            ${{github.workspace}}/artifactTests/packages/**/*.nupkg
+            ${{github.workspace}}/artifactTests/packages/**/*.snupkg
+   
+      - uses: actions/upload-artifact@v2
+        if: always()
+        name: "Upload test artifact"
+        with:
+          name: testResults
+          path: |
+            ${{ github.workspace }}/artifactTests/coverage/testResults/**/*.xml
+            ${{ github.workspace }}/artifactTests/coverage/testResults/
+  
+  report:
+    name: Report
+    runs-on: ubuntu-latest
+    needs: buildTest
+    if: always()  
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+        with:
+          path: ${{ github.workspace }}/Results
+      - name: Test Report
+        uses: dorny/test-reporter@v1
```

Comment:
test-reporter's docs suggests that this is probably going to have problems for folks sending PRs from forks of the repo, since the implicit github token used there won't have rights to create checks. [They document](https://github.com/dorny/test-reporter#recommended-setup-for-public-repositories) a workaround that involves moving this this "report" job out to a different trigger

Reply:
Moved to separate action